### DISCUSSION TITLE
✅ Private key on iPad

### DISCUSSION THREAD

**Discussion by n4turalZ0mbie** [isontheline/pro.webssh.net#461] [Q&A] [answered]
Hello, I am trying WebSSH on my iPad with a RSA private key {RSA 2048][SHA256] without passphrase issued on my server with:
ssh-keygen -t rsa -f ipad_rsa -C myname@ipad

The public key is added to authorized_keys and the key is imported to the app and added to the connection and I removed the password. I have tried with and without "Force Keyboard Interactive".

I get "Connection failed + Connection successful but cannot open session" (translated from Swedish, a bit confusing :) ) and the log on the server says:
sshd[id]: dispatch_protocol_error: type 90 seq 6 [preauth]

It works perfect with password only on my iPad and I can connect with this key from other clients. Am I missing something?

**Comment by isontheline** [comment]
Hello @n4turalZ0mbie 👋

Perhaps verbose logging will help you and me to find the root cause.

In order to enable verbose logging you can follow the steps :
1. iOS Settings
2. WebSSH
3. Settings
4. Log Level > Verbose
5. Restart WebSSH
6. Reproduce your action
7. Find the log files inside iOS > Files > WebSSH

Have you some time to try out?

Have a nice day ☀️

**Reply by n4turalZ0mbie** [reply]
Hello and thank you, this is my log:
2021/11/24 21:26:14:248 [VERBOSE] No prefered language. Use system language if available.
2021/11/24 21:26:14:253 [VERBOSE] Locale Identifier : sv_SE
2021/11/24 21:26:14:253 [VERBOSE] Device language : sv
2021/11/24 21:26:14:253 [VERBOSE] Selected language : sv
2021/11/24 21:26:14:740 [VERBOSE] Did become active [AppDelegate]
2021/11/24 21:26:14:748 [VERBOSE] No pin code set. WebSSH won't be locked.
2021/11/24 21:26:15:339 [DEBUG] Launch_APP_Count_16.1 : 8
2021/11/24 21:26:15:344 [DEBUG] Launch_APP_Count : 8
2021/11/24 21:26:29:742 [VERBOSE] Want to connect with a Private Key
2021/11/24 21:26:29:765 [VERBOSE] Private Key is encrypted : 0
2021/11/24 21:26:29:765 [VERBOSE] Load Private Key in OpenSSH Format
2021/11/24 21:26:29:765 [VERBOSE] Launching connection...
2021/11/24 21:26:29:766 [VERBOSE] Connecting to SSH...
2021/11/24 21:26:29:766 [VERBOSE] Launching connection with block...
2021/11/24 21:26:29:766 [VERBOSE] Won't dismiss prompt because no one is displayed.
2021/11/24 21:26:29:845 [DEBUG] Connected to : 192.168.*my-server*
2021/11/24 21:26:29:852 [DEBUG] Challenge Authentication Success!
2021/11/24 21:26:29:964 [VERBOSE] Dismissing prompt...
2021/11/24 21:26:31:740 [VERBOSE] Error : Anslutning lyckad men kan inte öppna en session
2021/11/24 21:26:36:630 [DEBUG] Application will resign active [AppDelegate]
2021/11/24 21:26:37:692 [DEBUG] Application Did Enter Background

I also put the server log on debug:
Nov 24 21:36:59 *my-servername* sshd[835]: debug1: Forked child 838.
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: Set /proc/self/oom_score_adj to 0
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: rexec start in 4 out 4 newsock 4 pipe 6 sock 7
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: inetd sockets after dupping: 3, 3
Nov 24 21:36:59 *my-servername* sshd[838]: Connection from 192.168.*my-ipad* port 50054 on 192.168.*my-server* port 22
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: Client protocol version 2.0; client software version PuTTY_Release_0.70
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: no match: PuTTY_Release_0.70
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: Local version string SSH-2.0-OpenSSH_7.6p1 Ubuntu-4ubuntu0.5
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: permanently_set_uid: 123/65534 [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: list_hostkey_types: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: SSH2_MSG_KEXINIT sent [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: SSH2_MSG_KEXINIT received [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: kex: algorithm: curve25519-sha256@libssh.org [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: kex: host key algorithm: ssh-rsa [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: expecting SSH2_MSG_KEX_ECDH_INIT [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: rekey after 134217728 blocks [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: SSH2_MSG_NEWKEYS sent [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: expecting SSH2_MSG_NEWKEYS [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: SSH2_MSG_NEWKEYS received [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: rekey after 134217728 blocks [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: KEX done [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: userauth-request for user *my-username* service ssh-connection method keyboard-interactive [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: attempt 0 failures 0 [preauth]
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: PAM: initializing for "*my-username*"
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: PAM: setting PAM_RHOST to "192.168.*my-ipad*"
Nov 24 21:36:59 *my-servername* sshd[838]: debug1: PAM: setting PAM_TTY to "ssh"
Nov 24 21:36:59 *my-servername* sshd[838]: dispatch_protocol_error: type 90 seq 6 [preauth]

**Reply by isontheline** [reply]
@n4turalZ0mbie Thank you a lot for your feedback 👍

Strange 🤨

Could you please disable "Force interactive keyboard" in order to check if verbose logs are different.

**Reply by n4turalZ0mbie** [reply]
Thanks @isontheline, it gave me more clues and I solved it! It could not match the key in authorized_keys so I deleted the key and imported it again. I bought your app, thanks for your support!

**Comment by n4turalZ0mbie** [comment, marked as answer]
Solved by deleting and importing the key in the app (not sure how to close this)

**Reply by isontheline** [reply]
Nice to see that the new import solves your issue with the use of private key 👍

Thank you a lot for your quick feedback 🙏
